SimulationAgent/SimulationRunner.cs

```diff
@@ -143,12 +154,26 @@ public class SimulationRunner : ISimulationRunner
         /// create an actor responsible for
         /// sending the telemetry.
         /// </summary>
-        public void Start(Services.Models.Simulation simulation)
+        public async void Start(Services.Models.Simulation simulation)
         {
             // Use `starting` to exit as soon as possible, to minimize the number 
             // of threads pending on the lock statement
-            if (this.starting || this.running) return;
+            if (this.starting || this.running)
+            {
```

Comment:
this code should be moved to Agent.cs instead of here.

Reply:
Also, is it safe to change the signature of the Start method to async?